cpp/src/robotick/framework/Engine.cpp

```
@@ -60,15 +67,46 @@ namespace robotick
+		if (rhs > max_size - lhs)
+			return false;
+		out = lhs + rhs;
+		return true;
+	}
+
+	static size_t max_align_for_type(size_t alignment)
+	{
+		return (alignment > alignof(max_align_t)) ? alignment : alignof(max_align_t);
+	}
+
+	static bool align_workloads_cursor_for_type(const TypeDescriptor& type, size_t& workloads_cursor)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)